### PR TITLE
sign integration

### PR DESCRIPTION
**Pull request by matiu**
integrates sign verification into the server.

### CODE REVIEW

File: lib/model/wallet.js [hunk 1]
```diff
@@ -26,6 +28,10 @@ function Wallet(opts) {
 Wallet.fromObj = function (obj) {
 	var x = new Wallet();
 
+  // TODO add sanity checks OR migration steps?
+  if (!obj.pubKey || !obj.m || !obj.n)
```
Comment (isocolsky) [inline lib/model/wallet.js]: I don't know about polluting the code with these checks. After all, wallet objects will always be retrieved from server-side db. Is it really possible to have one of these fields missing? I would focus more on checking against db tampering on sensitive areas like addresses and txs.

File: test/integration.js [hunk 2]
```diff
@@ -203,6 +203,7 @@ describe('Copay server', function() {
       server = new CopayServer({
         storage: storage,
       });
+      server._verifySignature = sinon.stub().returns(true);
```
Comment (isocolsky) [inline test/integration.js]: What do you think about not stubbing this but instead use real pubKeys + real signatures on these tests?

Comment (matiu) [inline test/integration.js]: Ack on that.

File: lib/model/wallet.js (hunk 1 above)
Comment (matiu) [inline lib/model/wallet.js]: Ok but we need to prevent (catch?) wrong arguments crash the server.

Comment (isocolsky) [inline lib/model/wallet.js]: I agree with that. Maybe we should rename fromObj to something like fromTrustedObj or fromDb and trust the API (i.e. createWallet, joinWallet, etc) to run the appropriate checks?

**Comment by matiu** [comment]
Updated with comments and adds fixes and tests to `verifyMessageSignature`

**Pull request merged**